### DISCUSSION TITLE
Locking yourself out from the admin panel

### DISCUSSION THREAD

**Discussion by TimHal** [payloadcms/payload#993] [Feature Requests & Ideas]
# Bug Report
It is possible to lock yourself out from the admin panel completely when using an auth-enabled collection as admin users which has a required relationship field.

## Expected Behavior
Payload should either
 - send a warning when parsing the payload config 
 - ignore the `required` relationship when creating the first user

## Current Behavior

Payload greets you with the /create-first-user page. The form requires you to choose an option for the required relationship field. Since no entities of the target collection exist none can be choosen and you are stuck.

## Possible Solution

- skip validation hooks for /create-first-user when it comes to relationship fields
- give out a configuration warning when starting payload

## Steps to Reproduce

1. Checkout the getting started payload template
2. Uncomment the `Examples` collection
3. alter the `Users` collection to 
```
const Users: CollectionConfig = {
  slug: 'users',
  auth: true,
  admin: {
    useAsTitle: 'email',
  },
  fields: [
    {
      name: 'example',
      type: 'relationship',
      relationTo: 'examples'
      hasMany: false,
      required: true,
    }
  ],
};

```
4. Start payload

## Detailed Description

Payload version: latest (and possibly anything before that)
Browser: any

**Comment by jmikrut** [comment]
Hey @TimHal — I think this is actually by design and would be fairly convoluted to fix if we wanted to allow the bypassing of relationship fields on the create-first-user operation.

I kinda feel like this is up to the dev to fix as the schema, having a required relationship, clearly dictates that users are required to have data in required relationship fields in order to create a user.

However, I think there are a few ways that we can avoid having this problem:

1. If you have a required relationship field on your users collection, you could seed data within Payload `onInit` to ensure that there are documents available within your relationship field to use. 
2. You could also write a custom `validate` function that uses `fetch` to check and see if there _**are**_ any relationships present, and if none present, then don't require a value within the relationship field. If there ARE relationships present, then validate normally. This would be pretty simple and would not require seeding.

What do you think? I'm going to convert this to a discussion for now but I really would love to know your thoughts!

**Comment by TimHal** [comment]
Hey @jmikrut thanks for the feedback! I agree that it should come down to fixing the schema, that's how I resolved the original issue too. 

I think seeding the database would be the best way to go and make a nice addition to payload even apart from this issue. This would also resolve another concern where the user might not want to expose the admin dashboard at all on a public server (only using payload for the api) but still needs to define some initial data.

I think the best way to do seeding right now is to just have a method for it somewhere after `payload.init` right? Perhaps it would be cool to have the seed data as part of the collection it is seeding instead and let payload handle it.
